get_active_hooks.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+"""
+    Report on Service & Web hooks for organization.
+"""
+import argparse
+import client
+import logging
+import urlparse
+import yaml
+
+"""
+    Lore:   swapping hook.test for hook.ping will cause repetition of the
+            actions.  In particular, a number of repos post to IRC channels
+            and/or bugs on commits, so expect comments to that effect.
+"""
+
+logger = logging.getLogger(__name__)
+
+
+def report_hooks(gh, org, active_only, unique_only, do_ping, yaml_out):
+    org_handle = gh.organization(org)
+    org_struct = org_handle.to_json()
+    repo_list = []
+    org_struct['repo_list'] = repo_list
+    unique_hooks = set()
+    msg = "Active" if active_only else "All"
+    for repo in org_handle.iter_repos():
+        repo_struct = repo.to_json()
+        hook_list = []
+        repo_struct['hook_list'] = hook_list
+        repo_list.append(repo_struct)
+        repo_hooks = set()
+        ping_attempts = ping_fails = 0
+        for hook in repo.iter_hooks():
+            hook_struct = hook.to_json()
+            hook_list.append(hook_struct)
+            # if hook.name == "web", then this is a web hook, and there can be
+            # several per repo. The unique part is the hook.config['url'], which
+            # may contain sensitive info (including basic login data), so just
+            # grab scheme, hostname, and port.
+            if hook.name != "web":
+                name = hook.name
+            else:
+                url = hook.config['url']
+                parts = urlparse.urlparse(url)
+                # port can be None, which prints funny, but is good enough for
+                # identification.
```

Comment:
True. If you want to make it valid you could always say `parts.port or 80`, though I admit that's still ugly.

Finally, this will work:
```
''.join([parts.scheme, '://', parts.hostname, ':%s' % parts.port if parts.port else ''])
```

Personal preference ;) but it also smells like there should be a convenience function for this somewhere.